src/main/java/com/icthh/xm/ms/configuration/repository/impl/IdpConfigRepository.java

```diff
@@ -46,13 +47,16 @@ public class IdpConfigRepository implements RefreshableConfiguration {
      */
     private final Map<String, Map<String, IdpPublicClientConfig>> idpClientConfigs = new ConcurrentHashMap<>();
 
-    /**
-     * In memory storage.
-     * Stores information about tenant IDP clients public configuration that currently in process.
-     * <p/>
-     * We need to store this information in memory to avoid corruption previously registered in-memory tenant clients config
-     */
-    private final Map<String, Map<String, IdpPublicClientConfig>> tmpValidIdpClientPublicConfigs = new ConcurrentHashMap<>();
+    private final UpdateJwkEventService jwksService;
+    private final ThreadPoolTaskScheduler scheduler = createScheduler();
+    private final Map<String, ScheduledFuture<?>> tasks = new ConcurrentHashMap<>();
+    private final UpdateJwkEventService updateJwkEventService;
+
+    private static ThreadPoolTaskScheduler createScheduler() {
```

Comment:
Is it reasonable to use xm-scheduler for refresh events?

Reply:
It`s impossible, because xm-scheduler require xm-config, xm-config  will require xm-scheduler to start